libs/wingcompiler/src/compile.ts

```diff
@@ -136,100 +137,89 @@ export async function compile(entrypoint: string, options: CompileOptions): Prom
   const targetdir = options.targetDir ?? join(dirname(entrypoint), "target");
-  const wingFile = resolve(entrypoint);
-  log?.("wing file: %s", wingFile);
-  const wingDir = resolve(dirname(wingFile));
+  const entrypointFile = resolve(entrypoint);
+  log?.("wing file: %s", entrypointFile);
+  const wingDir = resolve(dirname(entrypointFile));
   log?.("wing dir: %s", wingDir);
   const testing = options.testing ?? false;
   log?.("testing: %s", testing);
   const target = determineTargetFromPlatforms(options.platform);
-  const tmpSynthDir = resolveSynthDir(targetdir, wingFile, target, testing, true);
+  const tmpSynthDir = resolveSynthDir(targetdir, entrypointFile, target, testing, true);
   log?.("temp synth dir: %s", tmpSynthDir);
-  const synthDir = resolveSynthDir(targetdir, wingFile, target, testing);
+  const synthDir = resolveSynthDir(targetdir, entrypointFile, target, testing);
   log?.("synth dir: %s", synthDir);
   const workDir = resolve(tmpSynthDir, ".wing");
   log?.("work dir: %s", workDir);
 
-  // TODO: couldn't be moved to the context's since used in utils.env(...)
-  // in the future we may look for a unified approach
-  process.env["WING_TARGET"] = target;
-  process.env["WING_VALUES"] = options.value?.length == 0 ? undefined : options.value;
-  process.env["WING_VALUES_FILE"] = options.values;
-  process.env["WING_IS_TEST"] = testing.toString();
-  process.env["WING_PLATFORMS"] = resolvePlatformPaths(options.platform);
+  const nearestNodeModules = (dir: string): string => {
+    let nodeModules = join(dir, "node_modules");
+    while (!existsSync(nodeModules)) {
+      nodeModules = dirname(dirname(nodeModules));
 
-  const tempProcess: { env: Record<string, string | undefined> } = { env: { ...process.env } };
+      if (nodeModules === "/" || nodeModules.match(/^[A-Z]:\\/)) {
+        break;
+      }
 
-  tempProcess.env["WING_SOURCE_DIR"] = wingDir;
-  if (options.rootId) {
-    tempProcess.env["WING_ROOT_ID"] = options.rootId;
-  }
-  // from wingDir, find the nearest node_modules directory
-  let wingNodeModules = join(wingDir, "node_modules");
-  while (!existsSync(wingNodeModules)) {
-    wingNodeModules = dirname(dirname(wingNodeModules));
-
-    if (wingNodeModules === "/" || wingNodeModules.match(/^[A-Z]:\\/)) {
-      break;
+      nodeModules = resolve(nodeModules, "node_modules");
     }
 
-    wingNodeModules = resolve(wingNodeModules, "node_modules");
-  }
+    return nodeModules;
+  };
+
+  let compilerModuleDir = nearestNodeModules(__dirname);
+  let sdkModuleDir = nearestNodeModules(join(require.resolve("@winglang/sdk"), "..", "..", ".."));
+  let wingNodeModules = nearestNodeModules(wingDir);
+  let workDirNodeModules = nearestNodeModules(workDir);
 
-  tempProcess.env["WING_SYNTH_DIR"] = tmpSynthDir;
-  tempProcess.env["WING_NODE_MODULES"] = wingNodeModules;
+  const nodePaths = [compilerModuleDir, sdkModuleDir, wingNodeModules, workDirNodeModules];
```

Comment:
I'd add a comment explaining why we need to look up in all of these directories / why the default node module resolution isn't sufficient

Reply:
Figured out that all we need is to make sure the correct wingsdk is loaded, so I simplified things with a hack here (also has a comment) https://github.com/winglang/wing/pull/5247/files#diff-0c9cb50eae5ecfdb23433d8f48fb3d99a717429999343b0574c4766865569464R325